src/marten/db/field/slug.cr

```diff
@@ -14,11 +16,21 @@ module Marten
           @null = false,
           @unique = false,
           @index = true,
-          @db_column = nil
+          @db_column = nil,
+          @slugify : Symbol? = nil
         )
         end
 
+        macro check_definition(field_id, kwargs)
+          # No-op max_size automatic checks...
+        end
+
         def validate(record, value)
+          if slugify?(value)
+            slug = Core::Sluggable.generate_slug(record.get_field_value(slugify.not_nil!).to_s, max_size)
+            record.set_field_value(id, slug)
```

Comment:
This is a side effect: the `#validate` method is responsible for validating field values but here we are mutating the record. Ideally, this shouldn't be the case. 

What about making use of the [`#prepare_save`](https://github.com/martenframework/marten/blob/753d3f2182e07d78eb91020c76e2d827cc8626d3/src/marten/db/field/base.cr#L98-L103) method instead for assigning the generated slug?